### PR TITLE
Fix throw exception by `\,`

### LINKED ISSUE #139: \thinspace exception
Start the WpfMath.Example project.
Clear the starting formula in the textbox.
Enter "\\,"
An exception is thrown in TexPredefinedFormulaParser.cs

### PR DESCRIPTION
**Pull request by rstm-sf**
Fix throw exception by `\,` (#139 )
Add test for `\,`

- Closes #139.

### CODE REVIEW

**Comment by rstm-sf** [comment]
Or, there seems to be another solution -- make non-static

**Comment by ForNeVeR** [comment]
Honestly I fail to understand how is this supposed to fix any errors. Could you please explain? What was the initial issue?

**Comment by rstm-sf** [comment]
Initial issue was that a value is added on an existing key
https://github.com/ForNeVeR/wpf-math/blob/351e010a3e76a82001dc337e822d82ef8012321f/src/WpfMath/TexPredefinedFormulaParser.cs#L177

But these are just the consequences of another issue -- `argValueParsers` and `actionParsers`  is `static`
https://github.com/ForNeVeR/wpf-math/blob/351e010a3e76a82001dc337e822d82ef8012321f/src/WpfMath/TexPredefinedFormulaParser.cs#L18-L19

In case of parsing `\,`, we get into the next loop
https://github.com/ForNeVeR/wpf-math/blob/351e010a3e76a82001dc337e822d82ef8012321f/src/WpfMath/TexPredefinedFormulaParser.cs#L111-L122

where calling parse `\thinspace`
https://github.com/ForNeVeR/wpf-math/blob/351e010a3e76a82001dc337e822d82ef8012321f/src/WpfMath/Data/PredefinedTexFormulas.xml#L48-L50

in `CreateTeXFormulaParser.Parse(SourceSpan, XElement)`
https://github.com/ForNeVeR/wpf-math/blob/351e010a3e76a82001dc337e822d82ef8012321f/src/WpfMath/TexPredefinedFormulaParser.cs#L169-L170

and `CreateTeXFormulaParser.TempFormulas` (for `\,`) assigned the reference to the another dictionary. And, after returning the result of parsing `\thinspace`, the reference to another dictionary is returned back, and the _worker_ is over
https://github.com/ForNeVeR/wpf-math/blob/351e010a3e76a82001dc337e822d82ef8012321f/src/WpfMath/TexPredefinedFormulaParser.cs#L118

**Comment by rstm-sf** [comment]
@ForNeVeR, hello!

Do you think there is a better way to solve the problem, or did I explain it poorly?

**Comment by ForNeVeR** [comment]
@rstm-sf I am sorry for delay, I was rather occupied (happens to me triple a year these days), so I haven't properly reviewed these changes yet. Consider the ball on my side now.

File: src/WpfMath.Tests/TestResults/BoxTests.abbreviationThinspace.approved.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿{
+  "Children": [],
+  "Source": null,
```
[the remark this answers is not in the document]
Comment (rstm-sf) [reply src/WpfMath.Tests/TestResults/BoxTests.abbreviationThinspace.approved.txt]: While trying to insert this.source here

https://github.com/ForNeVeR/wpf-math/blob/1583baee2ac9f595a87b18395d5c44d40d1b1d02/src/WpfMath/TexFormulaHelper.cs#L208

We get `\thinspace`

[the remark this answers is not in the document]
Comment (ForNeVeR) [reply src/WpfMath.Tests/TestResults/BoxTests.abbreviationThinspace.approved.txt]: The stuff in `TexFormulaHelper` is pretty badly written w.r.t. `Source` information, feel free to improve it.

File: src/WpfMath/TexPredefinedFormulaParser.cs [hunk 2]
```diff
@@ -315,24 +314,33 @@ public override object Parse(string value, string type)
...
             {
+                this.SharedCacheFormulas = sharedCacheFormulas;
             }
 
-            public IDictionary<string, TexFormula> TempFormulas
+            public IDictionary<string, TexFormula> SharedCacheFormulas
             {
                 get;
-                set;
+                private set;
```
Comment (ForNeVeR) [inline src/WpfMath/TexPredefinedFormulaParser.cs]: The whole setter is actually unnecessary. Get-only properties are settable in constructors anyway.

**Comment by ForNeVeR** [comment]
Thanks for your contribution!

**Pull request merged**